omod/src/main/java/org/openmrs/module/owa/web/controller/AddAppController.java

```diff
@@ -74,7 +74,8 @@ public String upload(@RequestParam("file") MultipartFile file, HttpServletReques
                     message = messageSourceService.getMessage("owa.not_a_zip");
                     log.warn("App is not a zip archive");
                     uploadedFile.delete();
-                    return message;
+                    model.addAttribute("message", "App is not a zip archive");
```

Comment:
Yes error messages should always be taken from messages.properties. It also helps for multiple language implementation.